ecommerce/management/commands/generate_legacy_enrollment_codes.py

```diff
@@ -0,0 +1,106 @@
+"""
+Generates enrollment (discount) codes for legacy users. 
+
+This is for learners that have paid for a course but aren't enrolled in a 
+course run with a verified enrollment. The list is generated outside this 
+command and is passed to it as a CSV file. The format of the file is:
+
+```
+learner email,courserun readable id
+```
+
+and this command will generate single-use per-user discounts for each learner
+and course combination (if a product exists for it). 
+"""
+from django.core.management import BaseCommand
+import csv
+
+from courses.models import CourseRun, Course
+from ecommerce.models import Discount, DiscountProduct, UserDiscount
+from ecommerce.constants import REDEMPTION_TYPE_ONE_TIME, DISCOUNT_TYPE_PERCENT_OFF
+from users.models import User
+
+
+class Command(BaseCommand):
+    """
+    Generates enrollment codes for legacy paid learners
+    """
+
+    help = "Generates enrollment codes for legacy paid learners"
+
+    def add_arguments(self, parser) -> None:
+        parser.add_argument(
+            "input_file",
+            type=str,
+            help="Input file. See documentation for file format.",
+        )
+        parser.add_argument(
+            "output_file", type=str, help="File to write the results to."
+        )
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        output_codes = []
+
+        with open(kwargs["input_file"], newline="") as csvfile:
+            reader = csv.reader(csvfile, delimiter=",", quotechar="\\")
+
+            for row in reader:
+                user = User.objects.filter(email=row[0]).first()
+
+                if user is None:
+                    self.stderr.write(
+                        self.style.ERROR(
+                            f"Can't find account for user {row[0]}, skipping row"
+                        )
+                    )
+                    continue
+
+                courserun = CourseRun.objects.filter(courseware_id=row[1]).first()
+
+                if courserun is None:
+                    course = Course.objects.filter(readable_id=row[1]).first()
```

Comment:
I think here you might need to map the `course_id` from MM into `readable_id` in MITxOnline through this mapping model CourseId from micromasters_import/models.py.

Reply:
CourseId in micromasters_import/models.py is the auto-incremented primary key of the course model, not the string field https://rc.mitxonline.mit.edu/admin/micromasters_import/courseid/7/change/. Are we exporting the ID 7 to csv from https://micromasters-rc.odl.mit.edu/admin/courses/course/7/change/?